maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -144,37 +144,41 @@
     public static final File DEFAULT_GLOBAL_TOOLCHAINS_FILE =
         new File( System.getProperty( "maven.conf" ), "toolchains.xml" );
 
-    private static final String EXT_CLASS_PATH = "maven.ext.class.path";
+    public static final String EXT_CLASS_PATH = "maven.ext.class.path";
 
-    private static final String EXTENSIONS_FILENAME = ".mvn/extensions.xml";
+    public static final String EXTENSIONS_FILENAME = ".mvn/extensions.xml";
 
-    private static final String MVN_MAVEN_CONFIG = ".mvn/maven.config";
+    public static final String MVN_MAVEN_CONFIG = ".mvn/maven.config";
 
     public static final String STYLE_COLOR_PROPERTY = "style.color";
 
-    private ClassWorld classWorld;
+    protected static final Pattern LAST_ANSI_SEQUENCE = Pattern.compile( "(\u001B\\[[;\\d]*[ -/]*[@-~])[^\u001B]*$" );
 
-    private LoggerManager plexusLoggerManager;
+    protected static final String ANSI_RESET = "\u001B\u005Bm";
```

Comment:
I wonder whether these should be rather part of `MessageUtils` and not `MavenCli`.

Reply:
`MessageUtils` is not part of maven-core, so this would have to wait until it is provided.
I can refactor the split by line with color support into a separate method which we can then move to `MessageUtils`.  

Ideally, I'd rather use something like https://github.com/jline/jline3/blob/master/terminal/src/main/java/org/jline/utils/AttributedCharSequence.java#L401-L422 which handles things more safely, but that may be a bit overkill here.

Reply:
I've extracted the line splitting logic into a separate method which should be merged into `MessageUtils`.